src/monitorAnimationSpeed.js

```diff
@@ -0,0 +1,38 @@
+import React, {Component} from 'react'
+import collectFps from 'collect-fps'
+
+const noop = () => {}
+
+export default ({
+  threshold = 30,
+  fpsCollector = collectFps
+}) => (Target) => {
+  let endFPSCollection = noop
+
+  const handleStartFPSCollection = (component) => {
+    if (endFPSCollection !== noop) { endFPSCollection() }
+
+    endFPSCollection = fpsCollector()
+  }
+
+  class MonitorAnimationSpeed extends Component {
+    constructor () {
+      super()
+
+      this.state = {fps: 60}
+    }
+
+    render () {
+      return <Target
+        {...this.props}
+        onStartFPSCollection={handleStartFPSCollection}
```

Comment:
I was thinking more for when you are using it in a stateless component, but I'm mostly just messing with you. :P